src/vs/editor/contrib/referenceSearch/browser/referenceSearch.ts

```diff
@@ -7,7 +7,7 @@
 import * as nls from 'vs/nls';
 import { KeyCode, KeyMod } from 'vs/base/common/keyCodes';
 import URI from 'vs/base/common/uri';
-import { TPromise } from 'vs/base/common/winjs.base';
+import { PPromise, TPromise } from 'vs/base/common/winjs.base';
```

Comment:
Is `Holder` a pattern used by VS Code? Looks like a simple event emitter to me.
Don't you think it will be simpler to use the callback pattern everywhere instead of mixing with promises?